web-admin/src/features/billing/plans/selectors.ts

```diff
@@ -53,3 +59,59 @@ export function getNextBillingCycleDate(curEndDateRaw: string): string {
   const nextStartDate = curEndDate.plus({ day: 1 });
   return nextStartDate.toLocaleString(DateTime.DATE_MED);
 }
+
+export function getOrganizationUsageMetrics(
+  organization: string,
+): CreateQueryResult<UsageMetricsResponse> {
+  return derived(
+    [
+      createAdminServiceGetBillingProjectCredentials({
+        organization,
+      }),
+    ],
+    ([credsResp], set) => {
+      if (!credsResp.data) return;
+      return getUsageMetrics(
+        credsResp.data.runtimeHost,
+        credsResp.data.instanceId,
+        credsResp.data.accessToken,
+      ).subscribe(set);
+    },
+  );
+}
```

Comment:
Is it necessary to wrap this in a `derived` store? What about using a TanStack Query's `select` functionality?

Reply:
`getUsageMetrics` is actually a query with a custom query function. I can rename it to clarify. Lets keep it as is to make sure it gets cached.